### PR TITLE
[FUCK] Sanitize Loaded Languages!

### LINKED ISSUE #15232: Language point issues
## Testmerges:
This happened when we removed half the lanauges

## Reproduction:

If you had a language before that was removed, the language point is not refunded, and is permanently unusable.

### PR DESCRIPTION
**Pull request by RimiNosha**
## About The Pull Request
Uhhh, not me! I blame uh- um- TG, yea- they should've known someone was gonna mess with the language system!

<sub>/s</sub>

Closes #15232

## How This Contributes To The Skyrat Roleplay Experience
I BROKE STUFF AND NOW IT'S NOT!

## Changelog

:cl:
fix: Fixed invisible uninteractable languages on the prefs menu, as well as being able to be assigned unobtainable languages because of said bug.
/:cl:

### CODE REVIEW

File: modular_skyrat/master_files/code/modules/client/preferences/middleware/languages.dm [hunk 1]
```diff
@@ -36,6 +36,14 @@
 	)
 	var/list/name_to_language
 
+/datum/preference_middleware/languages/New(datum/preferences)
+	. = ..()
+	// Sanitize languages
+	for(var/datum/language/lang_path in src.preferences.languages.Copy())
```
Comment (Zonespace27) [inline modular_skyrat/master_files/code/modules/client/preferences/middleware/languages.dm]:
```suggestion
	for(var/datum/language/lang_path as anything in preferences.languages)
```

File: modular_skyrat/master_files/code/modules/client/preferences/middleware/languages.dm [hunk 2]
```diff
@@ -36,6 +36,14 @@
...
+		lang_path = new lang_path()
```
Comment (Zonespace27) [inline modular_skyrat/master_files/code/modules/client/preferences/middleware/languages.dm]:
```suggestion
```

File: modular_skyrat/master_files/code/modules/client/preferences/middleware/languages.dm [hunk 3]
```diff
@@ -36,6 +36,14 @@
...
+		if(lang_path.secret)
```
Comment (Zonespace27) [inline modular_skyrat/master_files/code/modules/client/preferences/middleware/languages.dm]:
```suggestion
		if(initial(lang_path.secret))
```

File: modular_skyrat/master_files/code/modules/client/preferences/middleware/languages.dm [hunk 4]
```diff
@@ -36,6 +36,14 @@
...
+			src.preferences.languages.Remove(lang_path)
```
Comment (Zonespace27) [inline modular_skyrat/master_files/code/modules/client/preferences/middleware/languages.dm]:
```suggestion
			preferences.languages.Remove(lang_path)
```

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Comment by theselfish** [comment]
We almost got the languages back.

**Review by GoldenAlpharex — APPROVED** [review]
Looks good.

**Pull request merged**